e2e-tests/insurance/cypress/e2e/journeys/declarations/modern-slavery/modern-slavery.spec.js

```diff
@@ -0,0 +1,194 @@
+import { field as fieldSelector, headingCaption, yesRadio, noRadio } from '../../../../../../pages/shared';
+import { modernSlaveryPage } from '../../../../../../pages/insurance/declarations';
+import { PAGES } from '../../../../../../content-strings';
+import { DECLARATIONS_FIELDS as FIELDS } from '../../../../../../content-strings/fields/insurance/declarations';
+import { FIELD_VALUES } from '../../../../../../constants';
+import { DECLARATIONS as DECLARATIONS_FIELD_IDS } from '../../../../../../constants/field-ids/insurance/declarations';
+import { INSURANCE_ROUTES } from '../../../../../../constants/routes/insurance';
+
+const CONTENT_STRINGS = PAGES.INSURANCE.DECLARATIONS.MODERN_SLAVERY;
+
+const {
+  ROOT: INSURANCE_ROOT,
+  DECLARATIONS: { MODERN_SLAVERY },
+} = INSURANCE_ROUTES;
+
+const {
+  MODERN_SLAVERY: { WILL_ADHERE_TO_ALL_REQUIREMENTS, HAS_NO_OFFENSES_OR_INVESTIGATIONS, IS_NOT_AWARE_OF_EXISTING_SLAVERY },
+} = DECLARATIONS_FIELD_IDS;
+
+const {
+  intro: { answerTheQuestions, guidingPrinciplesLink, ifYouSayNo },
+} = modernSlaveryPage;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Declarations - Modern slavery page - TODO EMS-4023', () => {
+  let referenceNumber;
+  let url;
+
+  before(() => {
+    cy.completeSignInAndGoToApplication({}).then(({ referenceNumber: refNumber }) => {
+      referenceNumber = refNumber;
+
+      url = `${baseUrl}${INSURANCE_ROOT}/${referenceNumber}${MODERN_SLAVERY}`;
+
+      cy.navigateToUrl(url);
+      cy.assertUrl(url);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  after(() => {
+    cy.deleteApplication(referenceNumber);
+  });
+
+  it('renders core page elements', () => {
+    cy.corePageChecks({
+      pageTitle: CONTENT_STRINGS.PAGE_TITLE,
+      currentHref: `${INSURANCE_ROOT}/${referenceNumber}${MODERN_SLAVERY}`,
+      backLink: `${INSURANCE_ROOT}/${referenceNumber}${MODERN_SLAVERY}#`,
+    });
+  });
+
+  describe('page tests', () => {
+    beforeEach(() => {
+      cy.navigateToUrl(url);
+    });
+
+    it('renders a heading caption', () => {
+      cy.checkText(headingCaption(), CONTENT_STRINGS.HEADING_CAPTION);
+    });
+
+    describe('intro', () => {
+      it('should render `answer the questions` copy', () => {
+        cy.checkText(answerTheQuestions(), CONTENT_STRINGS.INTRO.ANSWER_THE_QUESTIONS);
+      });
+
+      it('should render a `guiding principles` link', () => {
+        cy.checkLink(guidingPrinciplesLink(), '#', CONTENT_STRINGS.INTRO.GUIDING_PRINCIPLES_LINK.TEXT);
```

Comment:
could the # also come from GUIDING_PRINCIPLES_LINK.HREF?

Reply:
it's not wise to do that - it should come from `LINKS.EXTERNAL`. 

Otherwise, as we've learnt before - if the content string HREF is incorrect, the test will stil pass.